py-polars/polars/lazyframe/engine_config.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping
+
+    from rmm.mr import DeviceMemoryResource  # type: ignore[import-not-found]
```

Comment:
TODO: remove this ignore in a follow-up when CI can install the relevant cudf-polars packages.

Reply:
Can you explain what this is does? Will we only hit this when cudf is installed?

Reply:
This is so I can provide a type annotation for the `memory_resource` slot in the `GPUEngine` object. A `DeviceMemoryResource` is an abstraction for a device-side memory allocator, which we would like users to be able to configure.

Reply:
Ah, I see now that it is rapids memory manager. I am a bit concerned about import times here, that they will trigger if rmm is installed even if you don't run on GPU.

Reply:
Since this one is inside a `TYPE_CHECKING` block, I think it shouldn't trigger?

Reply:
Note that CI passed even though RMM is not importable in CI right now.